zipline/gens/tradesimulation.py

```diff
@@ -188,6 +188,10 @@ def inner_loop(dt_to_use):
                     perf_tracker_benchmark_returns[trading_day] = 0.001
                     # use the last dt as market close
                     yield self.get_message(trading_day)
+
+                    self.algo.portfolio_needs_update = True
```

Comment:
IIRC, found this when a unit test tried to access account, and the `needs_update` flag was initially False... the `_needs_update` fields may subject to removal, since some of the bottlenecks they were addressing may go away.